File: ZemljaSlova.Services/Database/TicketTypeTransaction.cs
﻿using System;
using System.Collections.Generic;

namespace ZemljaSlova.Services.Database;

public partial class TicketTypeTransaction
{
    public int Id { get; set; }

    public byte ActivityTypeId { get; set; }

    public int TicketTypeId { get; set; }

    public int Quantity { get; set; }

    public DateTime CreatedAt { get; set; }

    public int UserId { get; set; }

    public string? Data { get; set; }

    public virtual TicketType TicketType { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: UI/zemljaslova_desktop/lib/models/ticket_type_transaction.dart
import 'package:json_annotation/json_annotation.dart';
import '../widgets/inventory_screen.dart';

part 'ticket_type_transaction.g.dart';

@JsonSerializable()
class TicketTypeTransaction implements InventoryTransaction {
  final int? id;
  @override
  final int activityTypeId;
  final int ticketTypeId;
  @override
  final int quantity;
  @override
  final DateTime createdAt;
  final int userId;
  @override
  final String? data;

  TicketTypeTransaction({
    this.id,
    required this.activityTypeId,
    required this.ticketTypeId,
    required this.quantity,
    required this.createdAt,
    required this.userId,
    this.data,
  });

  factory TicketTypeTransaction.fromJson(Map<String, dynamic> json) => _$TicketTypeTransactionFromJson(json);

  Map<String, dynamic> toJson() => _$TicketTypeTransactionToJson(this);
}

enum ActivityType {
  stock(1),
  sold(2),
  remove(3),
  rent(4);

  const ActivityType(this.value);
  final int value;

  static ActivityType fromValue(int value) {
    return ActivityType.values.firstWhere(
      (type) => type.value == value,
      orElse: () => ActivityType.stock,
    );
  }
} 